### PR TITLE
Environment handling

### PR DESCRIPTION
**Pull request by AtR1an**
This PR adds

- Configs to create a bundled environment during the build process, that is installed alongside the extension so that users don't have to deal with Python environments to use the extension.
- A preference page for the extension that allows
   - Selection of an environment (bundled, conda or manual similar to the KNIME Python preference pages)
   - Creation of a Conda environment for CPU (all OS) and GPU (Windows and Linux)
   - Specification of a cache directory

### CODE REVIEW

**Review by ajbond — APPROVED** [review]

File: se.redfield.bert/src/se/redfield/bert/prefs/BundledEnvironmentConfigsObserver.java [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+				} catch (final Exception ex) {
+					return;
+				}
+				try {
+					setAvailableCondaEnvironments(availableEnvironments);
+					final CondaEnvironmentConfig environmentConfig = m_condaEnvironmentsConfig.getPython3Config();
+					environmentConfig.getEnvironmentDirectory().setStringValue(createdEnvironment.getDirectoryPath());
+					testPythonEnvironment(true);
+				} catch (Exception ex) {
+					// Ignore, we still want to configure and test the second environment.
```
Comment (chaubold) [inline se.redfield.bert/src/se/redfield/bert/prefs/BundledEnvironmentConfigsObserver.java]: what second environment?

File: se.redfield.bert/src/se/redfield/bert/prefs/BertPreferencePage.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import org.knime.python2.prefs.AbstractPythonPreferencePage;
+import org.knime.python3.PythonSourceDirectoryLocator;
+import org.knime.python3.scripting.nodes.prefs.BundledCondaEnvironmentConfig;
+
+import se.redfield.bert.prefs.MultiOptionEnvironmentCreator.CondaEnvironmentCreationOption;
+
+/**
+ * The preference page.
+ * 
+ * @author Alexander Bondaletov
```
Comment (chaubold) [inline se.redfield.bert/src/se/redfield/bert/prefs/BertPreferencePage.java]: I guess that should be Adrian :)

File: se.redfield.bert/src/se/redfield/bert/prefs/BertPreferenceInitializer.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import org.knime.python2.config.ManualEnvironmentsConfig;
+import org.knime.python2.config.PythonConfig;
+import org.knime.python2.config.PythonEnvironmentType;
+import org.knime.python2.config.PythonEnvironmentTypeConfig;
+import org.knime.python2.prefs.PythonPreferences;
+
+/**
+ * Preferences initializer.
+ * 
+ * @author Alexander Bondaletov
```
Comment (chaubold) [inline se.redfield.bert/src/se/redfield/bert/prefs/BertPreferenceInitializer.java]: Author = @AtR1an ? :)

File: se.redfield.bert/src/se/redfield/bert/prefs/CustomCondaEnvironmentsPreferencePanel.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+				envConfig.getPythonInstallationInfo(), //
+				envConfig.getPythonInstallationWarning(), //
+				envConfig.getPythonInstallationError(), //
+				envCreator, //
+				m_panel, //
+				openCreationDialog//
+		);
+		final GridData gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+//        gridData.horizontalIndent = 20;
```
Comment (chaubold) [inline se.redfield.bert/src/se/redfield/bert/prefs/CustomCondaEnvironmentsPreferencePanel.java]: remove?

File: se.redfield.bert/src/se/redfield/bert/prefs/CustomCondaEnvironmentsPreferencePanel.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021 Redfield AB.
```
Comment (chaubold) [inline se.redfield.bert/src/se/redfield/bert/prefs/CustomCondaEnvironmentsPreferencePanel.java:2]: what license? 2022?

File: se.redfield.bert/src/se/redfield/bert/prefs/MultiOptionEnvironmentCreator.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	void startEnvironmentCreation(String environmentName, CondaEnvironmentCreationStatus status,
+			CondaEnvironmentCreationOption creationOption) {
+		startEnvironmentCreation(environmentName, creationOption.m_pathToEnvYml, null, status);
+	}
+
+	List<CondaEnvironmentCreationOption> getCreationOptions() {
+		return m_options;
+	}
+
+	// TODO disable GPU on Mac
```
Comment (chaubold) [inline se.redfield.bert/src/se/redfield/bert/prefs/MultiOptionEnvironmentCreator.java]: wasn't that done in the list of available envs already?

File: knime.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+group_id: se.redfield
+name: bert
+author: Redfield AB
+vendor: Redfield AB
+description: "Redfield BERT Nodes"
+long_description: |
+  The extension provides the nodes that allow to use Google BERT Python libarary without coding.
+  Currently only classification for the texts is implemented.
```
Comment (chaubold) [inline knime.yml]: typo: remove "the"

File: knime.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+group_id: se.redfield
+name: bert
+author: Redfield AB
+vendor: Redfield AB
+description: "Redfield BERT Nodes"
+long_description: |
+  The extension provides the nodes that allow to use Google BERT Python libarary without coding.
```
Comment (chaubold) [inline knime.yml]: Suggestion:
This extension provides nodes that make the functionality of Google's BERT library available inside KNIME.

**Review by chaubold — CHANGES_REQUESTED** [review]
Looks good, my comments are just addressing some formal things that could be improved.

File: se.redfield.bert/src/se/redfield/bert/prefs/BundledEnvironmentConfigsObserver.java (hunk 1 above)
Comment (AtR1an) [reply se.redfield.bert/src/se/redfield/bert/prefs/BundledEnvironmentConfigsObserver.java]: There used to be one for Python 2 but that doesn't apply here.

File: se.redfield.bert/src/se/redfield/bert/prefs/CustomCondaEnvironmentsPreferencePanel.java (hunk 5 above)
Comment (AtR1an) [reply se.redfield.bert/src/se/redfield/bert/prefs/CustomCondaEnvironmentsPreferencePanel.java:2]: I just kept with the project settings here.

File: knime.yml [hunk 9]
```diff
@@ -4,10 +4,13 @@ author: Redfield AB
...
 description: "Redfield BERT Nodes"
 long_description: |
-  The extension provides the nodes that allow to use Google BERT Python libarary without coding.
-  Currently only classification for the texts is implemented.
+  This extension provides nodes that make the functionality of Google's BERT library available in the KNIME Analytics Platform.
+  Currently, the extension is allows the training of text classification models as well as creating embeddings for texts.
   This extension is built on top of the TensorFlow 2 Python package. 
   It is also possible to execute BERT on GPU (nVidia CUDA-based).
+  NOTE: The installation of this extension can fail on Windows if the installation folder of your Analytics Platform has a long path.
+  This is due to Window's long path limitations. You can circumvent this problem by enabling long path support as outlined here:
```
Comment (chaubold) [inline knime.yml]: typo: Windows'

**Review by chaubold — APPROVED** [review]

**Pull request merged**